File: 0796 Rotate String/0796.py
class Solution:
    def rotateString(self, s: str, goal: str) -> bool:
        if len(s) != len(goal):
            return False
        else:
            def valid(i):
                for j in range(len(goal)):
                    if s[i] != goal[j]:
                        return False
                    i = (i + 1) % len(s)
                return True

            for i in range(len(s)):
                if valid(i):
                    return True
            return False

File: 0796 Rotate String/0796.cpp
class Solution {
public:
    bool valid(int i, string s, string goal){
        for (int j = 0 ; j < goal.size(); j++){
            if (s[i] != goal[j]){
                return false;
            }
            i = (i + 1) % s.size();
        }
        return true;
    }

    bool rotateString(string s, string goal) {
        if (s.size() != goal.size()){
            return false;
        } else {
            for (int i = 0; i < s.size(); i++){
                if (valid(i, s, goal)){
                    return true;
                }
            }
            return false;
        }
    }
};
